test/rc-test.js

```diff
@@ -116,18 +125,36 @@ test('using --tag-prefix will set the tag prefix', function (t) {
 })
 
 function runRc (t, args, env, cb) {
-  var cmd = 'node ' + path.resolve(__dirname, '..', 'rc.js') + ' ' + args
-  env = Object.assign({}, process.env, env)
-  exec(cmd, { env: env }, function (err, stdout, stderr) {
-    t.error(err, 'no error')
-    t.equal(stderr.length, 0, 'no stderr')
-    var result
-    try {
-      result = JSON.parse(stdout.toString())
-      t.pass('json parsed correctly')
-    } catch (e) {
-      t.fail(e.message)
+  var pkg = {
+    name: 'test',
+    private: true,
+    scripts: {
+      install: 'node ' + path.resolve(__dirname, '..', 'rc.js') + ' ' + args
     }
-    cb(result)
+  }
+
+  var tmp = tempy.directory()
+  var json = JSON.stringify(pkg)
+
+  fs.writeFile(path.join(tmp, 'package.json'), json, function (err) {
+    if (err) throw err
+
+    var npm = process.platform === 'win32' ? 'npm.cmd' : 'npm'
+    var cmd = npm + ' run install'
+
+    env = Object.assign(cleanEnv(process.env), env)
+
+    exec(cmd, { env, cwd: tmp }, function (err, stdout, stderr) {
+      t.error(err, 'no error')
+      t.equal(stderr.trim(), '', 'no stderr')
+
+      try {
+        var result = JSON.parse(stdout.slice(stdout.indexOf('{')))
```

Comment:
Ideally we'd use `--silent` to suppress npm's output (before our JSON) but npm 7 suppresses all output: https://github.com/npm/cli/issues/1958